com.unity.render-pipelines.universal/Runtime/ShaderDebugPrintManager.cs

```diff
@@ -0,0 +1,264 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Mathematics;
+using UnityEngine.Assertions;
+
+namespace UnityEngine.Rendering.Universal
+{
+    public sealed class ShaderDebugPrintManager
+    {
+        private static readonly ShaderDebugPrintManager instance = new ShaderDebugPrintManager();
+
+        private const int DebugUAVSlot = 7;
```

Comment:
I suppose a check should be done against https://docs.unity3d.com/ScriptReference/SystemInfo-supportedRandomWriteTargetCount.html

Reply:
Yes. At least a check is a good idea. This part is indeed a bit of a hack :) 
Ideally we would control this fully programmatically, but how to affect the corresponding shader definition. 
Is there a way to control both from code?

Reply:
I don't think there's a way to have this configurable in the shader...
Looks like you can at least distinguish between "MRT4 supported" and "MRT8 supported" using the following syntax:
"#pragma require mrt8 : MY_MRT8_KEYWORD"

Reply:
Yeah, I guess that would need some sort of codgen or magic macro for "last".
That pragma is good info.

Reply:
I haven't found a good solution for this, maybe change the slot automatically (defines) depending on the platform.